middlewares/auth.middleware.js

```diff
@@ -0,0 +1,65 @@
+import jwt from 'jsonwebtoken';
+import {
+  UnAuthorizedError,
+  NotFoundError,
+  ForbiddenError,
+} from '../helpers/errors.js';
+import { roles } from '../constants/index.js';
+import User from '../models/user/index.js';
+
+const SECRET_KEY = process.env.SECRET_KEY;
+
+/**
+ * Middleware for role-based JWT token verification.
+ *
+ * @param {Object} options - Middleware options.
+ * @param {string} options.role - Required user role ('user', 'chef', 'courier', 'admin').
+ *
+ * @example
+ * // Protect a route requiring 'chef' role
+ * router.get('/protectedRoute', verifyToken({ role: 'chef' }), async (req, res) => {
+ *   // Your protected route logic here
+ * });
+ *
+ * @returns {string} - Required roleId ('userId', 'chefId', etc.) if the token is valid.
+ */
+
+const verifyToken = ({ role }) => {
+  const getTokenFromHeaders = (req) => req.headers.authorization?.split(' ')[1];
+
+  const getRoleId = (userRoles) =>
+    userRoles.find((userRole) => userRole.name === role)?.id;
+
+  return async (req, res, next) => {
+    try {
+      const token = getTokenFromHeaders(req);
+      if (!token) throw new UnAuthorizedError();
+
+      const { id } = jwt.verify(token, SECRET_KEY);
+
+      const user = await User.findById(id);
```

Comment:
better to use `.exec()` on mongoose queries
https://mongoosejs.com/docs/promises.html#queries-are-not-promises

Reply:
Thanx for info! I'll fix it

Reply:
Fixed 
https://github.com/Autumn-fairytale/AFT-back/pull/34/files#diff-8018e43a672f37668be0b873bd5cb66267eba0256327671fa14313fe8ff17bb9R50